### ISSUE TITLE
ecs-agent crashed - Unable to discover poll endpoint after registering instance with ECS

### ISSUE THREAD

**Issue by abhishek-tamrakar** [aws/amazon-ecs-agent#882]
```
2017-07-11T15:28:17Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:28:37Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:37910->10.10.0.2:53: i/o timeout
2017-07-11T15:28:37Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:37910->10.10.0.2:53: i/o timeout
2017-07-11T15:28:37Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:28:37Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:37910->10.10.0.2:53: i/o timeout
2017-07-11T15:28:39Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:28:59Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:58205->10.10.0.2:53: i/o timeout
2017-07-11T15:28:59Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:58205->10.10.0.2:53: i/o timeout
2017-07-11T15:28:59Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:28:59Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:58205->10.10.0.2:53: i/o timeout
2017-07-11T15:29:04Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:29:24Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:53286->10.10.0.2:53: i/o timeout
2017-07-11T15:29:24Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:53286->10.10.0.2:53: i/o timeout
2017-07-11T15:29:24Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:29:24Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:53286->10.10.0.2:53: i/o timeout
2017-07-11T15:29:33Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:29:53Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:48205->10.10.0.2:53: i/o timeout
2017-07-11T15:29:53Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:48205->10.10.0.2:53: i/o timeout
2017-07-11T15:29:53Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:29:53Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:48205->10.10.0.2:53: i/o timeout
2017-07-11T15:30:09Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:30:14Z [ERROR] Error getting message from ws backend: error: [read tcp 10.10.0.241:55486->54.239.20.177:443: read: connection reset by peer], message: [], messageType: [-1]
2017-07-11T15:30:14Z [DEBUG] Reconnecting to ACS in: 256.648809ms
2017-07-11T15:30:14Z [DEBUG] Received connect to ACS message
2017-07-11T15:30:29Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:42519->10.10.0.2:53: i/o timeout
2017-07-11T15:30:29Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:42519->10.10.0.2:53: i/o timeout
2017-07-11T15:30:29Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:30:29Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:42519->10.10.0.2:53: i/o timeout
2017-07-11T15:30:34Z [DEBUG] Reconnecting to ACS in: 394.574314ms
2017-07-11T15:30:34Z [DEBUG] Received connect to ACS message
2017-07-11T15:30:54Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:40518->10.10.0.2:53: i/o timeout
2017-07-11T15:30:54Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:40518->10.10.0.2:53: i/o timeout
2017-07-11T15:30:54Z [DEBUG] Reconnecting to ACS in: 592.875266ms
2017-07-11T15:30:55Z [DEBUG] Received connect to ACS message
2017-07-11T15:31:15Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:48451->10.10.0.2:53: i/o timeout
2017-07-11T15:31:15Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:48451->10.10.0.2:53: i/o timeout
2017-07-11T15:31:15Z [DEBUG] Reconnecting to ACS in: 969.553689ms
2017-07-11T15:31:16Z [DEBUG] Received connect to ACS message
2017-07-11T15:31:26Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:33147->10.10.0.2:53: i/o timeout
2017-07-11T15:31:26Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:33147->10.10.0.2:53: i/o timeout
2017-07-11T15:31:26Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:31:26Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:33147->10.10.0.2:53: i/o timeout
2017-07-11T15:31:36Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:60312->10.10.0.2:53: i/o timeout
2017-07-11T15:31:36Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:60312->10.10.0.2:53: i/o timeout
2017-07-11T15:31:36Z [DEBUG] Reconnecting to ACS in: 1.423050732s
2017-07-11T15:31:37Z [DEBUG] Received connect to ACS message
2017-07-11T15:31:57Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:40471->10.10.0.2:53: i/o timeout
2017-07-11T15:31:57Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:40471->10.10.0.2:53: i/o timeout
2017-07-11T15:31:57Z [DEBUG] Reconnecting to ACS in: 2.233671524s
2017-07-11T15:31:59Z [DEBUG] Received connect to ACS message
2017-07-11T15:32:19Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:57641->10.10.0.2:53: i/o timeout
2017-07-11T15:32:19Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:57641->10.10.0.2:53: i/o timeout
2017-07-11T15:32:19Z [DEBUG] Reconnecting to ACS in: 3.16423117s
2017-07-11T15:32:23Z [DEBUG] Received connect to ACS message
2017-07-11T15:32:31Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:32:43Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:56209->10.10.0.2:53: i/o timeout
2017-07-11T15:32:43Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:56209->10.10.0.2:53: i/o timeout
2017-07-11T15:32:43Z [DEBUG] Reconnecting to ACS in: 4.93901278s
2017-07-11T15:32:48Z [DEBUG] Received connect to ACS message
2017-07-11T15:32:51Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:34589->10.10.0.2:53: i/o timeout
2017-07-11T15:32:51Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:34589->10.10.0.2:53: i/o timeout
2017-07-11T15:32:51Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:32:51Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:34589->10.10.0.2:53: i/o timeout
2017-07-11T15:33:08Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:56974->10.10.0.2:53: i/o timeout
2017-07-11T15:33:08Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:56974->10.10.0.2:53: i/o timeout
2017-07-11T15:33:08Z [DEBUG] Reconnecting to ACS in: 6.991768398s
2017-07-11T15:33:15Z [DEBUG] Received connect to ACS message
2017-07-11T15:33:35Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:38077->10.10.0.2:53: i/o timeout
2017-07-11T15:33:35Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:38077->10.10.0.2:53: i/o timeout
2017-07-11T15:33:35Z [DEBUG] Reconnecting to ACS in: 10.102846876s
2017-07-11T15:33:45Z [DEBUG] Received connect to ACS message
2017-07-11T15:33:52Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:34:05Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:54287->10.10.0.2:53: i/o timeout
2017-07-11T15:34:05Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:54287->10.10.0.2:53: i/o timeout
2017-07-11T15:34:05Z [DEBUG] Reconnecting to ACS in: 16.09757645s
2017-07-11T15:34:12Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:51751->10.10.0.2:53: i/o timeout
2017-07-11T15:34:36Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:51751->10.10.0.2:53: i/o timeout
2017-07-11T15:34:36Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:34:36Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:51751->10.10.0.2:53: i/o timeout
2017-07-11T15:34:36Z [DEBUG] Received connect to ACS message
2017-07-11T15:34:56Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:49378->10.10.0.2:53: i/o timeout
2017-07-11T15:34:56Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:49378->10.10.0.2:53: i/o timeout
2017-07-11T15:34:56Z [DEBUG] Reconnecting to ACS in: 21.883141243s
2017-07-11T15:35:18Z [DEBUG] Received connect to ACS message
2017-07-11T15:35:38Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:33113->10.10.0.2:53: i/o timeout
2017-07-11T15:35:38Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:33113->10.10.0.2:53: i/o timeout
2017-07-11T15:35:38Z [DEBUG] Reconnecting to ACS in: 37.813433515s
2017-07-11T15:35:42Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:36:02Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:44897->10.10.0.2:53: i/o timeout
2017-07-11T15:36:02Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:44897->10.10.0.2:53: i/o timeout
2017-07-11T15:36:02Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:36:02Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:44897->10.10.0.2:53: i/o timeout
2017-07-11T15:36:16Z [DEBUG] Received connect to ACS message
2017-07-11T15:36:36Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:41505->10.10.0.2:53: i/o timeout
2017-07-11T15:36:36Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:41505->10.10.0.2:53: i/o timeout
2017-07-11T15:36:36Z [DEBUG] Reconnecting to ACS in: 54.585598491s
2017-07-11T15:37:02Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:37:22Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:42419->10.10.0.2:53: i/o timeout
2017-07-11T15:37:22Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:42419->10.10.0.2:53: i/o timeout
2017-07-11T15:37:22Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:37:22Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:42419->10.10.0.2:53: i/o timeout
2017-07-11T15:37:30Z [DEBUG] Received connect to ACS message
2017-07-11T15:38:23Z [DEBUG] Connecting to TCS endpoint https://ecs-t-1.us-east-1.amazonaws.com/
2017-07-11T15:38:41Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:57367->10.10.0.2:53: i/o timeout
2017-07-11T15:38:41Z [ERROR] Error connecting to ACS: , dial tcp: lookup ecs-a-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:57367->10.10.0.2:53: i/o timeout
2017-07-11T15:38:41Z [DEBUG] Reconnecting to ACS in: 1m17.630126396s
2017-07-11T15:38:43Z [WARN] Error creating a websocket client: dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:45873->10.10.0.2:53: i/o timeout
2017-07-11T15:38:43Z [ERROR] Error connecting to TCS: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:45873->10.10.0.2:53: i/o timeout
2017-07-11T15:38:43Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:38:43Z [INFO] Error from tcs; backing off: , dial tcp: lookup ecs-t-1.us-east-1.amazonaws.com on 10.10.0.2:53: read udp 10.10.0.241:45873->10.10.0.2:53: i/o timeout
2017-07-11T15:39:33Z [INFO] Saving state! module="statemanager"
2017-07-11T15:39:33Z [ERROR] Error adding container reference to image state: no such image
2017-07-11T15:39:33Z [DEBUG] Cannot add empty image state module="dockerstate"
2017-07-11T15:39:33Z [INFO] Saving state! module="statemanager"
2017-07-11T15:39:33Z [DEBUG] Sending container change event to tcs, container: , status: PULLED
2017-07-11T15:39:33Z [DEBUG] Still waiting for module="TaskEngine" map="map[]"
2017-07-11T15:39:33Z [DEBUG] Event stream ContainerChange received events, broadcasting to listeners...
2017-07-11T15:39:33Z [DEBUG] Ignoring event for container, id: , status: 1
2017-07-11T15:39:33Z [INFO] Saving state! module="statemanager"
2017-07-11T15:39:33Z [DEBUG] Still waiting for module="TaskEngine" map="map[]"
2017-07-11T15:39:33Z [DEBUG] Sending container change event to tcs, container: , status: STOPPED
2017-07-11T15:39:33Z [DEBUG] Sending container change event to tcs, container: , status: STOPPED
2017-07-11T15:39:33Z [DEBUG] Event stream ContainerChange received events, broadcasting to listeners...
2017-07-11T15:39:33Z [DEBUG] Event stream ContainerChange received events, broadcasting to listeners...
2017-07-11T15:39:33Z [DEBUG] Could not map container to task, ignoring, err: Could not map docker id to task: , id:
2017-07-11T15:39:33Z [DEBUG] TaskHandler, Waiting on semaphore to send...
2017-07-11T15:39:33Z [DEBUG] TaskHandler, Aquiring lock for sending event...
2017-07-11T15:39:33Z [DEBUG] TaskHandler, Aquired lock!
2017-07-11T15:39:33Z [DEBUG] Could not map container to task, ignoring, err: Could not map docker id to task: , id:
```
And
```
2017-07-11T15:44:43Z [ERROR] Error connecting to ACS: InactiveInstanceException: Instance 3735e957-76bb-4fb7-a0d0-171c6f39ac1e is inactive
...
2017-07-11T15:45:45Z [ERROR] Unable to discover poll endpoint: %!(EXTRA *awserr.baseError=RequestError: send request failed
2017-07-11T15:45:45Z [INFO] Error from tcs; backing off: RequestError: send request failed
```
And
```
2017-07-11T15:46:56Z [ERROR] Error getting message from ws backend: error: [websocket: close 1008 (policy violation): InvalidContainerInstance: Missing container instance arn], message: [], messageType: [-1]
2017-07-11T15:46:56Z [DEBUG] Unsubscribing event handler TCSDeregisterContainerInstanceHandler from event stream DeregisterContainerInstance
2017-07-11T15:46:56Z [INFO] Error from tcs; backing off: websocket: close 1008 (policy violation): InvalidContainerInstance: Missing container instance arn
```
I have almost the same issue that caused a downtime of around 8 mins, the task came up on same instance after that.

The last one came when the instance was already registered with ECS cluster, I dont want this to be repeated, please suggest the way out of this kind of issue. 
We are not suing any proxy but all traffic via NAT.

**Comment by abhishek-tamrakar** [comment]
The issue https://github.com/aws/amazon-ecs-agent/issues/676 was related but had no solution, So i created this issue. Please help.

**Comment by adnxn** [comment]
Hey @abhishek-tamrakar - So our Metrics endpoint, TCS, does indicate there is an issue but this is not what is causing the problem you are seeing. 

The log entries after `15:34:41` are expected, as your container instance was deregistered from the perspective of ECS.

We can also see that the websocket connection to our Agent Communication endpoint, ACS, was abruptly closed at `15:30:14`. This leads us to believe that the problem you are experiencing was a result of intermittent network issues. Specifically the `read udp 10.10.0.241:48205->10.10.0.2:53: i/o timeout` messages are indicative of DNS resolution failing. 

We suggest spreading instances in your cluster across AZs to ensure that that task placements can continue as expected if an AZ is experiencing intermittent networking issues.

**Comment by abhishek-tamrakar** [comment]
Thanks, @adnxn for replying back. this was indeed an intermittent DNS resolution issue. I would also like to ask for a suggestion how to hande it at launch time if there is a way. 
There is something else that happened at the same time. Let me show you few more logs. I can send you the one with actual id's on your email if that helps in any case, please let me know:

A new task was Started and then the old instance got deregistered from ELB (expected) and it happened with the newest instance too (unexpected) before it even confirms the service registered in ELB. If that's not it, the old task has begun draining. This is not as expected behavior from ECS AFAIK with minimum healthy percentage=100% and max=200% and no of instances 1 + no of tasks running it should have waited unless the new task registers itself in ELB before beginning to drain the old instance (Correct me if my understanding is wrong here but I reproduced this myself many times with the same setup).
Unfortunately, I don't see much logs resembling this behavior in ecs-agent logs but these events are from the same deployment and timestamp can confirm this. Now if this happens even when I have instances in multi-AZ and network is able to pull images. I saw this happening many times as of now the new service registers in ELB, marked healthy and then as old service drains, the new crashes and comes up again after some time. Kindly let me know your thoughts in this, I can try getting more logs if you need.
```
    "message": "(service Nginx) registered 1 instances in (elb xxxxxxxxx)", 
                    "id": "xxxxxxxxxxxxxxxxxxxx", 
                    "createdAt": 1499787598.505 -> # GMT: Tuesday, July 11, 2017 3:39:58.505 PM
                }, 
                {
                    "message": "(service Nginx) has started 1 tasks: (task xxxxxxxxxxxxxxxxxxxxxxxxx).", 
                    "id": "xxxxxxxxxxxxxxxxxxxxxxxxxxx", 
                    "createdAt": 1499787490.262
                }, 
                {
                    "message": "(service Nginx) was unable to place a task because no container instance met all of its requirements. The closest matching (container-instance xxxxxxxxxxxxxxxxxxxxxxxxxxx) encountered error \"memberOf constraint unsatisfied\". For more information, see the Troubleshooting section of the Amazon ECS Developer Guide.", 
                    "id": "xxxxxxxxxxxxxxxxxxxxxxxxxxx", 
                    "createdAt": 1499787298.843 -> # We had only 1 task and an additional instance where the new task was created.
                }, 
                {
                    "message": "(service Nginx) has begun draining connections on 1 tasks.", 
                    "id": "xxxxxxxxxxxxxxxxxxxxxxxxxxx", 
                    "createdAt": 1499787044.688
                }, 
                {
                    "message": "(service Nginx) deregistered 1 instances in (elb xxxxxxxxx)", 
                    "id": "xxxxxxxxxxxxxxxxxxxxxxxxxxx", 
                    "createdAt": 1499787044.68
                }, 
                {
                    "message": "(service Nginx) deregistered 1 instances in (elb xxxxxxxxx)", 
                    "id": "xxxxxxxxxxxxxxxxxxxxxxxxxxx", 
                    "createdAt": 1499787044.674
                }, 
                {
                    "message": "(service Nginx) has started 1 tasks: (task xxxxxxxxxxxxxxxxxxxxxxxxxxx).", 
                    "id": "xxxxxxxxxxxxxxxxxxxxxxxxxxx", 
                    "createdAt": 1499786762.664 --> # GMT: Tuesday, July 11, 2017 3:26:02.664 PM 
                }, 
```

**Comment by adnxn** [comment]
> this was indeed an intermittent DNS resolution issue.

okay great, we got that one identified =]

> Correct me if my understanding is wrong here but I reproduced this myself many times with the same setup

you're correct, that should be the expected behavior.

> Let me show you few more logs. I can send you the one with actual id's on your email if that helps in any case, please let me know

yes please send them to adnkha at amazon dot com, having the actual id's would help us investigate this further. thanks !

**Comment by adnxn** [comment]
I'm closing this issue for now since we haven't heard back. Feel free to reopen or comment if you have some more information (or incase I missed your email !) about your specific use case. Thanks.
